### DISCUSSION TITLE
Scheduled Blog Posts?

### DISCUSSION THREAD

**Discussion by cameronpresley** [squidfunk/mkdocs-material#5370] [Ideas]
Hi all!

I've been using the blog plugin for the last two months or so, and I really enjoy the ease of creating new blog posts (especially over my previous solution).

That being said, one piece of functionality I'd love to see is the ability to schedule a post. In my previous solution (WordPress), I was able to create a post but then specify a date/time when the post would be published and accessible by others. When I write posts, I typically write multiple posts at once, but stagger the publishing of them (e.g. I might write all the posts for a month over a weekend, and then will have each one get posted on Monday morning).

Doing some digging, the closest thing I've found is the `draft` attribute of the post (see below)

However, that doesn't prevent a user from navigating to the page directly. 

To mimic this workflow, I can create a PR with the new post, but won't merge the PR until it's ready to go live. This works, but it's cumbersome as I would need to create multiple PRs and then remember to merge them once a week.

I'm interested in hearing what the community thinks about this new functionality as I believe the blog plugin is still new and I'm the new person around here.

Looking forward to the conversations!

**Comment by kamilkrzyskow** [comment]
Hi, I don't have access to the Blog plugin, as I'm not an Insiders user, just responding in general.

MkDocs is a static site generator, there is no back-end server, which would allow for scheduling logic, which would withhold on publishing a draft post or prevent the user from accessing the page altogether. The site is built once and then served as is to the end user.
Wordpress is different in that regard because it is a PHP driven software, which requires a back-end server to begin with.

A quick solution I see would be staging PRs, like you said, and then you could add some kind of [scheduling](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onschedule) to your Publish workflow, and then use some kind of [automerge action](https://github.com/marketplace/actions/merge-pull-requests-automerge-action) to merge PRs based on some logic. I think setting up a scheduling back-end with GitHub actions is possible, but I don't know much more that those links above ✌️ 

I'm not sure how the drafts functionality works, but I guess another solution would require squidfunk to serve "encrypted" articles, which would be loaded in JavaScript after the scheduled date. The site url would be still available before the scheduled date, but the article wouldn't show up. However, I consider this feature unlikely to be introduced, because the theme is open source, so any client side "encryption" could be bypassed anyways.

**Reply by cameronpresley** [reply]
Hi @kamilkrzyskow!

Yep, the more I think about it and read through your response, the more that makes sense. Originally, I was thinking of tying into the 404 logic (e.g. if the route doesn't exist, return 404, if the route does exist, but the scheduled time hasn't passed, then return 404), but the lack of the server or anything to intercept the request makes that not possible.

Given that I'm hosting on GitHub Pages, I would need something at that layer to intercept the call and do this logic (which is something that GitHub Pages doesn't support).

I agree that the whole encryption thing is not ideal, and I would hate to see that happen, as that would add a ton of complexity without much value.

Thank you for taking the time to respond and helping me think about things differently, I definitely appreciate it!
